### PR TITLE
V5: YAML support

### PR DESCRIPTION
**Pull request by qjebbs**
The PR:

- Brings YAML support, for `v2ray run`, `v2ray test`, `v2ray convert` (external loader) 
- Removes `merge` command, because `convert` takes the job

`v2ray help convert`:
```
usage: v2ray convert [c1.json] [<url>.json] [dir1] ...

Merge and convert V2Ray config files.

Arguments:

    -i, -input
        Input format, options: "json", "yaml". 
        Default "json"

    -o, -output
        Output format, options: "json", "yaml", "protobuf"/"pb".
        Default "json"

    -r
        Load confdir recursively.

Examples:

    v2ray convert -output=protobuf config.json           (1)
    v2ray convert -output=yaml config.json               (2)
    v2ray convert -input=yaml config.yaml                (3)
    v2ray convert "path/to/dir"                          (4)
    v2ray convert -i yaml -o protobuf c1.yaml <url>.yaml (5)

(1) Convert json to protobuf
(2) Convert json to yaml
(3) Convert yaml to json
(4) Merge json files in dir
(5) Merge yaml files and convert to protobuf

Use "v2ray help config-merge" for more information about merge.
```

### CODE REVIEW

File: main/distro/all/all.go [hunk 1]
```diff
@@ -61,6 +61,12 @@ import (
 	// The following line loads JSON internally
 	_ "v2ray.com/core/main/jsonem"
 
+	// YAML config support. Choose only one from the two below.
+	// The following line loads YAML from v2ctl
+	// _ "v2ray.com/core/main/yaml"
+	// The following line loads YAML internally
+	_ "v2ray.com/core/main/yamlem"
+
```
Comment (kslr) [inline main/distro/all/all.go:69]: 我们不需要再考虑v2ctl了

Comment (qjebbs) [reply main/distro/all/all.go:69]: 我想就顺手做了，跟json一样，默认不启用。这样要裁剪的人还有机会用yaml

Comment (kslr) [reply main/distro/all/all.go:69]: 接下来是打算完全剥离 ctl, 最终移除掉

**Comment by qjebbs** [comment]
最新的提交移除了外部加载器，现在裁剪的用户只能使用pb格式（可能多数情况也是这样，否则v2ray+v2ctl尺寸比裁剪前大），如果有必要，把v2ctl的相关代码也删除

The extenal loaders are removed in last commit, cropping v2ray users now have to use `.pb` only (It's how it happens in most cases, since v2ray cropped+v2ctl have larger size). If needed, I can remove the v2ctl codes also.

**Review by DuckSoft — APPROVED** [review]
LGTM

File: commands/all/convert.go [hunk 2]
```diff
@@ -2,70 +2,129 @@ package all
...
 Arguments:
 
+	-i, -input
+		Input format, options: "json", "yaml". 
+		Default "json"
+
+	-o, -output
+		Output format, options: "json", "yaml", "protobuf"/"pb".
+		Default "json"
+
```
Comment (database64128) [inline commands/all/convert.go:32]:
```
	-i, --input
		Specify the input format.
		Available values: "json", "yaml"
		Default: "json"
	-o, --output
		Specify the output format
		Available values: "json", "yaml", "protobuf", "pb"
		Default: "json"
```

**Review by database64128 — CHANGES_REQUESTED** [review]
Instead of using `-i` and `-o` to specify input or output file format, it might be a better idea to just have a syntax like `v2ray convert <input_file> <output_file>`. Use file extensions to determine source and target file format.

Examples:

- `v2ray convert config.json config.yaml`: converts `config.json` to `config.yaml`.
- `v2ray convert test.jsonc test.yml`: converts `test.jsonc` to `test.yml`.

**Comment by qjebbs** [comment]
> Instead of using `-i` and `-o` to specify input or output file format, it might be a better idea to just have a syntax like `v2ray convert <input_file> <output_file>`. Use file extensions to determine source and target file format.
> 
> Examples:
> 
> * `v2ray convert config.json config.yaml`: converts `config.json` to `config.yaml`.
> * `v2ray convert test.jsonc test.yml`: converts `test.jsonc` to `test.yml`.

Better way is to work like `run` command, to guess the format if not specified. But:

- I don't want to make things complicate now. Maybe later someone could enhance it
- For same reason, convert now only outputs to stdout: `v2ray convert ... > out.yml`
- `-output`, `-input` are always needed, pipe output won't give you chance to guess format

**Review by kslr — APPROVED** [review]

**Comment by kslr** [comment]
Thanks for your work.

**Pull request merged**